src/pages/journal-vouchers/forms/JournalVoucherForm.js

```diff
@@ -20,31 +18,44 @@ import CustomTextArea from 'src/components/Inputs/CustomTextArea'
 
 import { GeneralLedgerRepository } from 'src/repositories/GeneralLedgerRepository'
 import { SystemRepository } from 'src/repositories/SystemRepository'
+import { useForm } from 'src/hooks/form'
+import useDocumentType from 'src/hooks/dcRefBhv'
+import { useError } from 'src/error'
 
-export default function JournalVoucherForm({ labels, maxAccess, recordId }) {
+export default function JournalVoucherForm({ labels, access, recordId, general = {} }) {
   const [isLoading, setIsLoading] = useState(false)
   const [editMode, setEditMode] = useState(!!recordId)
   const [responseValue, setResponseValue] = useState(null)
+  const { stack: stackError } = useError()
+
+  const {
+    query: { data },
+    maxAccess: maxAccess,
+    onChangeNra
+  } = useDocumentType({
+    functionId: SystemFunction.JournalVoucher,
+    access: access
+  })
 
   const [initialValues, setInitialData] = useState({
     recordId: null,
     reference: '',
     date: new Date(),
     notes: '',
     currencyId: '',
-    dtId: '',
+    dtId: data?.dtId,
     status: 1,
     rateCalcMethod: 1,
     exRate: 1
   })
-
   const { getRequest, postRequest } = useContext(RequestsContext)
 
   const invalidate = useInvalidate({
     endpointId: GeneralLedgerRepository.JournalVoucher.qry
   })
 
-  const formik = useFormik({
+  const { formik } = useForm({
+    maxAccess,
     initialValues,
```

Comment:
Set here directly.

Reply:
it lead to make loop

Reply:
We have to find a solution for this. This might be common moving forward.